### PR TITLE
Issue/quick 124 get high xl apdex for real

### PR DESCRIPTION
**Pull request by dagguh**

### CODE REVIEW

**Comment by dagguh** [comment]
Spinning: https://server-gdn-bamboo.internal.atlassian.com/browse/QUICK-JHWR27-1

**Comment by dagguh** [comment]
https://s3-eu-west-1.amazonaws.com/quicksilver-jhwr-cache-ireland/QUICK-124-cherry-picked-v2/jira-exploration-chart.html
`#64: TESTED: 16, SKIPPED: 45, FAILED: 3, REMAINING: 0`
Good progress, but we still `java.lang.Exception: We have nothing to recommend`

2 of these fails were caused by failed Jira distro download.
1 was caused by failed JDK install.
Both are flakes, rerunning to fill in the fails: https://server-gdn-bamboo.internal.atlassian.com/browse/QUICK-JHWR27-2

File: src/test/kotlin/com/atlassian/performance/tools/hardware/tuning/HeapTuning.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    private fun bumpHeap(
+        desiredGigabytes: Int,
+        hardware: Hardware
+    ): JiraJvmArgs {
+        val maxGigabytes = when (hardware.jira) {
+            C52xlarge -> 12
+            C54xlarge -> 25
+            C48xlarge -> 50
+            C59xlarge -> 60
```
Comment (kema2000) [inline src/test/kotlin/com/atlassian/performance/tools/hardware/tuning/HeapTuning.kt:44]: I like that we have different heap per instance type. I think anything over 50G might not be necessary. We have not seen much GC problem with 50G according to Jprofiler result.

Comment (dagguh) [reply src/test/kotlin/com/atlassian/performance/tools/hardware/tuning/HeapTuning.kt:44]: Note that this is `maxGigabytes`.
We actually `coerceAtMost(desiredGigabytes)`, so we'll use 50 GB even though more is possible.

**Comment by dagguh** [comment]
So the rerun indeed shows good perf, as we can see in the [chart|https://s3-eu-west-1.amazonaws.com/quicksilver-jhwr-cache-ireland/QUICK-124-cherry-picked-v2/jira-exploration-chart.html] and we've got recommendations:
```
Recommending [HardwareTestResult(hardware=[7 x c5.18xlarge Jira, m4.4xlarge DB], apdex=0.7601193624981833, apdexes=[0.7593216944286811, 0.7609170305676856], httpThroughput=271.394305655877 over PT1S, httpThroughputs=[266.60053407612463 over PT1S, 276.18807723562935 over PT1S], errorRate=0.018748898269264336, errorRates=[0.018382593879022215, 0.019115202659506458]), HardwareTestResult(hardware=[6 x c5.18xlarge Jira, m4.4xlarge DB], apdex=0.7460252733250117, apdexes=[0.7462797619047619, 0.7457707847452616], httpThroughput=276.46560560682167 over PT1S, httpThroughputs=[276.51652349356544 over PT1S, 276.41468772007784 over PT1S], errorRate=0.018597050841023657, errorRates=[0.01821369436057271, 0.01898040732147461])]
```

There are still failures:
Jira: `#64: TESTED: 23, SKIPPED: 38, FAILED: 3, REMAINING: 0`
DB: `#8: TESTED: 2, SKIPPED: 0, FAILED: 6, REMAINING: 0`

But we show high XL apdex. It was good to merge 👍

**Pull request merged**